api/end_device.proto

```diff
@@ -316,20 +315,27 @@ message MACState {
   // Regenerated on each downlink.
   repeated MACCommand pending_requests = 10;
 
+  message JoinRequest {
+    DLSettings downlink_settings = 6 [(gogoproto.nullable) = false, (validate.rules).message.required = true];
+    RxDelay rx_delay = 7 [(validate.rules).enum.defined_only = true];
+    CFList cf_list = 8 [(gogoproto.customname) = "CFList"];
+  }
```

Comment:
Maybe reserve the old fields, and add a comment that while these fields are actually part of the accepted join, they're in here for compatibility reasons.